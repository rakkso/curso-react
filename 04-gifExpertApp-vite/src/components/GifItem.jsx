import PropTypes from 'prop-types';

export default function GifItem( {title, url, id}) {
  return (
    <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
        GitItem</div>
  )
}

GifItem.protoTypes ={
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

