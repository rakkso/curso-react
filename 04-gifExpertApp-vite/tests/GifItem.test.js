import { render } from "@testing-library/react"
import GifItem from "../src/components/GifItem"

describe('Pruebas de GifItem', () => {
    
    test('Debe hacer match con el snapshot', () => { 
        
    const titulo ='Titulo ejemplo'
    const url = 'urlejemplo'

    
    const {container } = render (<GifItem title={titulo} url={url}/>);

    expect (container).toMatchSnapshot();
    })
    
    

})