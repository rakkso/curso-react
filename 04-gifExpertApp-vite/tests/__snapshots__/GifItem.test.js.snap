// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo ejemplo"
      src="urlejemplo"
    />
    <p>
      Titulo ejemplo
    </p>
    GitItem
  </div>
</div>
`;
